codacy/templates/fluentd/fluentd-conf.yaml

```diff
@@ -14,28 +14,28 @@ data:
     </match>
 
     # Remove unneeded fields from the logs
-    <filter **>
+    <filter kubernetes.var.log.containers.**{{ .Release.Namespace }}**>
       @type record_modifier
       <record>
-        for_remove ${record["kubernetes"].delete("namespace_name"); record["kubernetes"].delete("pod_id"); record["kubernetes"].delete("namespace_labels"); record["kubernetes"].delete("container_info"); record["kubernetes"].delete("labels"); record["kubernetes"].delete("container_name"); record["kubernetes"].delete("pod_name"); record["kubernetes"].delete("container_image_id"); record["kubernetes"].delete("host"); record.delete("container_info"); record.delete("docker")}
+        for_remove ${record["kubernetes"].delete("namespace_name"); record["kubernetes"].delete("pod_id"); record["kubernetes"].delete("namespace_labels"); record["kubernetes"].delete("container_info"); record["kubernetes"].delete("labels"); record.delete("container_name"); record.delete("pod_name"); record["kubernetes"].delete("container_image_id"); record["kubernetes"].delete("host"); record.delete("container_info"); record.delete("docker")}
```

Comment:
for some reason deleting the record[kubernetes][pod_name] has a knock on effect on record[pod_name] which becomes null. The same happens for the container_name field.

In turn, if both of them are null the output path to S3 gets messed up, ending up in something like `logs#//2020-10-01.json`, which is invalid.

Reply:
So instead we keep the kubernetes podname and containername, and remove the other fields instead